src/Symfony/Component/Validator/Constraints/FileValidator.php

```diff
@@ -225,6 +225,13 @@ private function factorizeSizes(int $size, int|float $limit, bool $binaryFormat)
             $coefFactor = self::KB_BYTES;
         }
 
+        // If $limit < $coef, $limitAsString could be < 1 with less than 3 decimals.
+        // In this case, we would end up displaying an allowed size < 1 (eg: 0.1 MB).
+        // It looks better to keep on factorizing (to display 100 kB for example).
+        while ($limit < $coef) {
```

Comment:
Should we check that `1 < $limit` before entering since?
One could provide eg `$constraint->maxSize`

Reply:
If `$constraint->maxSize = 0` then the file size is not validated at all and it never reaches that code. I'll add a test on 4.4 since I already opened a PR.

Reply:
See https://github.com/symfony/symfony/pull/45511/files#diff-6f05554980834dc353ca1d8a7baedac277ab2aa7b00f33ffd246c9909799fe39R188

Reply:
What about `$constraint->maxSize = -1`?

Reply:
@fancyweb Any news?

Reply:
-1 is impossible (as well as any negative value) since it cannot pass `File::normalizeBinaryFormat()`.